src/instruction/instructions/subtract/subs.ts

```diff
@@ -23,23 +24,28 @@ export class SubsRegistersInstruction extends BaseInstruction {
   private rdPattern: string = '0001101000000XXX'
   private rnPattern: string = '0001101000XXX000'
   private rmPattern: string = '0001101XXX000000'
-  private minExpectedOptionCount: number = 2
-  private maxExpectedOptionCount: number = 3
+  private expectedOptionCountMin: number = 2
+  private expectedOptionCountMax: number = 3
 
   public canEncodeInstruction(name: string, options: string[]): boolean {
     let immCheckIndex = options.length === 3 ? 2 : 1
     return (
       super.canEncodeInstruction(name, options) &&
       isOptionCountValid(
         options,
-        this.minExpectedOptionCount,
-        this.maxExpectedOptionCount
+        this.expectedOptionCountMin,
+        this.expectedOptionCountMax
       ) &&
       !isImmediate(options[immCheckIndex])
     )
   }
 
   public encodeInstruction(options: string[]): Halfword[] {
+    checkOptionCount(
```

Comment:
Again I believe this check is redundant as it will never throw an exception because the code here will never be reached because the canEncodeInstruction method will not return true.

Reply:
Same argument as before and does not hurt in my opinion